utils.py

```diff
@@ -211,62 +211,109 @@ def saveImage(window):
     overlays = window.overlays
     exthandler = window.exthandler
     Tk().withdraw()
-    file = filedialog.asksaveasfile(
-        filetypes=[("PNG Image", "*.png")], defaultextension=".png"
+    file = filedialog.asksaveasfilename(
+        filetypes=[("PNG Image", "*.png"),("TIFF Image", "*.tiff"),("CSV File", "*.csv")]
     )
 
     if file:
-        filename = file.name
-
-        if (
-            messagebox.askquestion(
-                "Save options", "Do you want to save with the annotations?"
+        filename = file.split("/")[-1]
+
+        if filename.endswith('.tiff'):
+            if (messagebox.askquestion("Before we proceed",f"Are you sure you want to export image as {filename}?") == "yes"):
+                tiffExport(window, file)
+            else:
+                messagebox.showinfo("Export Cancelled", f"You have cancelled the process of exporting {filename}")
+
+        elif filename.endswith('.csv'):
+            if (messagebox.askquestion("Before we proceed",f"Are you sure you want to export image as {filename}?") == "yes"):
+                csvExport(window, file)
+            else:
+                messagebox.showinfo("Export Cancelled", f"You have cancelled the process of exporting {filename}")
+
+        else:
+            print(file)
+            pngExport(window, file, imageSurface, overlays, exthandler)
+
+def pngExport(window, filename, imageSurface, overlays, exthandler):
+    """Function to export image as .PNG format
+
+    Args:
+        window (class Window object): object that refers to main window
```

Comment:
You needn't have "class Window object", you can just mention <library>.Window :)

Reply:
Done.